src/Canvas.js

```diff
@@ -17,62 +17,64 @@ const HAS_MATRIX = (function (div) {
 })(document.createElement('div'))
 
 export default class Canvas {
-  constructor(img, opts) {
+  constructor(element, opts) {
     // make this object observable
     o(this)
     this.opts = opts
-    this.img = img
-    this.wrapper = img.parentNode
+    this.element = element
+    this.wrapper = element.parentNode
     this.isLoaded = false
 
-    // store the initial image properties - deep clone
-    this.initial = img.cloneNode(true)
+    // store the initial element properties - deep clone
+    this.initial = element.cloneNode(true)
   }
   /**
-   * Load the image
+   * Load the element
    * @returns { Object } - Canvas
    */
   load() {
+    const isImage = this.element.complete !== undefined
+    const isImageReady = isImage && this.element.width && this.element.height && this.element.complete
 
-    if (!this.img.width || !this.img.height || !this.img.complete) {
-      this.img.onload = () => this.onImageLoaded()
+    if (isImage && !isImageReady) {
+      this.element.onload = () => this.onElementLoaded()
```

Comment:
I would still keep the `this.element.width || !this.element.height || !this.element.complete`
and otherwise fallback to the async loading
```js
this.element.onload = this.element.oncanplay = () => this.onElementLoaded()
```

Reply:
Do you want to keep this so that only images and video are possible elements to apply the parallax to because I think doing it my way it should work with other elements and this still checks for `!this.element.width || !this.element.height || !this.element.complete` just in the variable.

Reply:
I would first fix the unit test and then probably check if the parallax element needs an async loading like for images or videos `['IMG', 'VIDEO'].includes(this.element.tagName)` and otherwise trigger the `onElementLoaded` right the way

Reply:
Sorry @GianlucaGuarini I have been away from this over Christmas.  I made a small change that checks for image or video and attached the onElementLoaded function to complete or oncanplay accordingly, if it is neither then onElementLoaded is triggered immediately.